src/components/tools/use-image-content-url.ts

```diff
@@ -8,15 +8,10 @@ type OnUrlChangeFn = (url: string, filename?: string, context?: IDocumentContext
 
 export function useImageContentUrl(content: ImageContentModelType, onUrlChange: OnUrlChangeFn) {
   const context = useContext(DocumentContextReact);
-  const syncedChanges = useRef(0);
   useEffect(() => {
-    const dispose = autorun(() => {
-      if (content.changeCount > syncedChanges.current) {
-        const { url, filename } = content;
-        url && onUrlChange(url, filename, context);
-        syncedChanges.current = content.changeCount;
-      }
-    });
+    const dispose = reaction(() =>
+      ({ url: content.url, filename: content.filename }), ({ url, filename }) =>
+        url && onUrlChange(url, filename, context));
```

Comment:
Given that the arguments are two separate functions, I think it would be clearer to format it more like:
```suggestion
    const dispose = reaction(
      () => ({ url: content.url, filename: content.filename }),
      ({ url, filename }) => url && onUrlChange(url, filename, context));
```
if formatting it on multiple lines.